### DISCUSSION TITLE
How to use nativewind with @rneui/themed ?

### DISCUSSION THREAD

**Discussion by jakesee** [react-native-elements/react-native-elements#3800] [Q&A] [answered]
I am using nativewind and Expo Go for development and testing, and I have have the following code snippet:

```typescript
import { View, Text } from "react-native"
import { Text as Text2 } from "@rneui/themed"

<View>
    <Text className="text-red-600">Email</Text>
    <Text2 className="text-red-600">Email</Text2>
</View>
``` 

I expect both text to appear red in color according to the nativewind utility. However, only the react-native Text is in red and Text2 is in black.

Is it possible to use nativewind with react-native-elements at all? If so, how can I do it?

Thanks in advance.

**Comment by keygun-development** [comment]
@jakesee did you figure this out?

**Comment by utkarsh1097** [comment, marked as answer]
https://www.nativewind.dev/guides/custom-components#passing-styles-to-components

Unless third-party component libraries (like RNE) explicitly implement the support for it, `className` does not get passed down to the child native component. RNE is just a wrapper over the actual native components. You can refer to the [source file of each component](https://github.com/react-native-elements/react-native-elements/blob/next/packages/base/src/Text/Text.tsx) to understand better.

To support Nativewind, we need to explicitly pass down any additional props (like `className`) to the child component. This needs to be added as a new enhancement PR. You can refer to https://www.nativewind.dev/v4/guides/third-party-components for more details.

**Reply by theianmay** [reply]
Updated link: https://www.nativewind.dev/v5/guides/third-party-components
